File: src/OK.Hookman.Core/Models/HookModel.cs
using System;
using System.Collections.Generic;
using OK.Hookman.Core.Enums;

namespace OK.Hookman.Core.Models
{
    public class HookModel
    {
        public int Id { get; set; }
        public int EventId { get; set; }
        public StatusEnum StatusId { get; set; }
        public string Data { get; set; }
        public string Message { get; set; }
        public string RequestUrl { get; set; }
        public IDictionary<string, string> RequestHeaders { get; set; }
        public string RequestBody { get; set; }
        public int? ResponseCode { get; set; }
        public IDictionary<string, string> ResponseHeaders { get; set; }
        public string ResponseBody { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime? UpdatedDate { get; set; }

        public EventModel Event { get; set; }
        public SenderModel Sender { get; set; }
        public StatusModel Status { get; set; }
    }
}

File: src/OK.Hookman.UI/static/src/app/_models/hook.model.ts
import { EventModel } from './event.model';
import { StatusModel } from './status.model';
import { SenderModel } from './sender.model';

export class HookModel {
    constructor(
        public id: number,
        public eventId: number,
        public senderId: number,
        public statusId: number,
        public data: string,
        public message: string,
        public requestUrl: string,
        public requestHeaders: {[key: string]: string},
        public requestBody: string,
        public responseCode: number,
        public responseHeaders: {[key: string]: string},
        public responseBody: string,
        public event: EventModel,
        public sender: SenderModel,
        public status: StatusModel,
        public createdDate: Date,
        public updatedDate: Date
    ) { }
}
